api/v1beta2/network_types.go

```diff
@@ -254,6 +266,9 @@ type VPCSpec struct {
 	// Defaults to 10.0.0.0/16.
 	CidrBlock string `json:"cidrBlock,omitempty"`
 
+	// IPAMPool defines the IPAM pool to be used for VPC.
+	IPAMPool *IPAMPool `json:"ipamPool,omitempty"`
```

Comment:
I think we should probably rename this so its specific to IPV4 and then also add a IPAMPool field in the [IPv6](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/f43a36bfbe4284aa5f4e8c0b1feac857945adbfb/api/v1beta2/network_types.go#L234) struct.

Reply:
Or probably better the suggestion from @Skarlso 

```yaml
network: 
  vpc:
    ipam:
       ipv6Block: bla
       ipv4Block: blah
```

Reply:
We also need to think about the fact we want to completely evolve the networking side of CAPA as the base types have long assumed IPv4 and so what would the ideal structure be for IPv4 and IPv6 fields. 

We introduced the **IPv6** type (thanks @Skarlso), so do we think we want to also have an **IPv4** type as well going forward?

wdyt @Skarlso @MaxFedotov @AndiDog ?

Reply:
I think the VPC/IPAM section would be more intuitive if it bundled both options. I can understand if you would like to do that in a separate PR though. But I wouldn't want to separate these. The IPv6 section is rather an indicator to enable ipv6 ATM. The IPAM as it is now conveys the feeling to set both.

Let's do the following. Let's add the ability to set ipv6 in a followup but add the fields now. But set a warning or an error if someone does set ipv6 IPAM to say, it's not yet supported. That way, the followup PR can then be added since the fields are already there.

WDYT?

Reply:
> I think the VPC/IPAM section would be more intuitive if it bundled both options. I can understand if you would like to do that in a separate PR though. But I wouldn't want to separate these. The IPv6 section is rather an indicator to enable ipv6 ATM. The IPAM as it is now conveys the feeling to set both.

I agree that we need to have 2 separate configurations - one for `ipv4` and one for `ipv6`, because the user may use IPAM pools for both, and in order to do it he has to create 2 different pools and they will have different IDs.

But from my POV, as we already have some `ipv6` related configurations in separate `ipv6` struct (like `poolId` or `cidrBlock`), it will be quite properly from UX side to put IPAM configuration here as well. 

And also we have to keep in mind, that when using `ipv6`, the user is not able to specify both `IPv6 CIDR owned by me`(`poolId` in `ipv6` struct) and `ipam pool`. So keeping them in the same struct will make this grouping more logical.

Reply:
That does make sense. Okay put it under the IPv6 tree then. I'm just cautious here that we don't start piling stuff in both sections of that makes sense.

Reply:
Putting it in `type IPv6` makes sense, and we should document in the field/struct comments which fields are mutually exclusive (e.g. `IPv6.CidrBlock` and the new `IPv6.IPAMPool*`, I guess). Also, having some fields already added now, but unsupported for the moment, seems a good idea given that they have clear code comments.